solution.cpp

```diff
@@ -10,125 +10,212 @@
- *   through the pointer.
- * - It might be better to return the array from the function rather than pass
- *   the array and modify it, but the syntax is relatively uncommon, and might
- *   be confusing.
- */
-void sieve(const int size, bool primes[]);
-
-/**
- * Read numbers from `infilename`, and if they are prime, output them to
- * `outfilename`, separated by newlines.
- *
- * Arguments:
- * - `size`: The size of (the number of elements in) the array.
- * - `primes`: An array of boolean values, with each element set to `true` if
- *   its index is prime, and `false` otherwise.
- * - `infilename`: A string representing the file path to read from.
- * - `outfilename`: A string representing the file path to write to.
- *
- * Returns:
- * - 0 on success, or -1 on error.
- */
-int write_primes( const int size,
-                  bool primes[],
-                  const string& infilename,
-                  const string& outfilename );
-
-/**
- * Read numbers from `infilename`, and if they are composite, output them to
- * `outfilename`, separated by newlines.
- *
- * Arguments:
- * - `size`: The size of (the number of elements in) the array.
- * - `primes`: An array of boolean values, with each element set to `true` if
- *   its index is prime, and `false` otherwise.
- * - `infilename`: A string representing the file path to read from.
- * - `outfilename`: A string representing the file path to write to.
- *
- * Returns:
- * - 0 on success, or -1 on error.
- */
-int write_composites( const int size,
-                      bool primes[],
-                      const string& infilename,
-                      const string& outfilename );
-
-// ----------------------------------------------------------------------------
-
-int main() {
-    int max = find_max("input.txt");
-    if (max == -1) {
-        cout << "ERROR in `find_max()`" << endl;
-        return 1;  // error
-    }
-
-    bool primes[max+1];
-    sieve(max+1, primes);
-
-    int ret;  // for storing return values, to check for error codes
-    //
-    ret = write_primes(max+1, primes, "input.txt", "primes.txt");
-    if (ret == -1) {
-        cout << "ERROR in `write_primes()`" << endl;
-        return 1;  // error
+using std::ifstream; 
+using std::ofstream; 
+
+
+
+int main()
+{
+	int num[] = {1};
+	int t,z;
+	ifstream myfile("Text.txt");
+	
+	// below counting code received from Matt
+    if (myfile.is_open())
+    {
+        
+        t = 0;
+        while (!myfile.eof()){
+            
+            myfile >> z;
+            t++;
+        }
+        cout<< t<<endl;
     }
-    //
-    ret = write_composites(max+1, primes, "input.txt", "composites.txt");
-    if (ret == -1) {
-        cout << "ERROR in `write_composites()`" << endl;
-        return 1;  // error
-    }
-
-    return 0;  // success
+    myfile.close();
+    
+	ifstream inFile;
+	inFile.open("Text.txt");
+	for (int i = 0; i <= t; i++)
+	{
+		inFile >> num[i];
+	}
+	
+	inFile.close();
+
+	int myPrime[] = {0};
```

Comment:
same thing here, with `myPrime[]` as with `num[]` -- this was only the first project, and it's been a long time since it was turned in, so there's no point in changing it now, but, in the future, please *never do this*.  thanks :-)